pkg/experiment/metastore/compaction/compactor/compaction_queue.go

```diff
@@ -248,21 +246,55 @@ func (s *stagedBlocks) delete(block string) blockEntry {
 	ref.batch.size--
 	s.stats.blocks.Add(-1)
 	if ref.batch.size == 0 {
-		s.queue.removeBatch(ref.batch)
+		// This is the last block in the batch, so we want to remove it
+		if ref.batch != s.batch {
```

Comment:
This is the actual fix: we should never ever try to delete the staging batch from the queue – it hasn't been flushed and added to the queue yet.